src/Catty/Functions/Math/LnFunction.swift

```diff
@@ -0,0 +1,44 @@
+/**
+ *  Copyright (C) 2010-2018 The Catrobat Team
+ *  (http://developer.catrobat.org/credits)
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  An additional term exception under section 7 of the GNU Affero
+ *  General Public License, version 3, is available at
+ *  (http://developer.catrobat.org/license_additional_term)
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see http://www.gnu.org/licenses/.
+ */
+
+class LnFunction: SingleParameterFunction {
+    
+    static var tag = "LN"
+    static var name = "ln"
+    static var defaultValue = -Double.infinity
+    static var requiredResource = ResourceType.noResources
+    static var isIdempotent = true
+    static let position = 40
+    
+    static func firstParameter() -> FunctionParameter {
+        return .number(defaultValue: 0)
+    }
+    
+    func value(parameter: AnyObject?) -> Double {
+        guard let value = parameter as? Double else { return type(of: self).defaultValue }
+        return log(value)
```

Comment:
`ln('abc')` will evaluate to -inf but should evaluate to 0. Please do not forget to add tests for these cases.